### PR TITLE
polarArray with object rotation

### LINKED ISSUE #118: Polar array support with changing orientation
Currently there is no easy way to achieve the right result:

afbeelding

Wheras it is trival to get to the left one:
```python
r1 = cq.Workplane("XY" )\
    .box(10, 10, 1).faces(">Z")\
    .polarArray(3.5,0,360,6).rect(1.6,1.6).cutThruAll()
```

I think the right scenario occurs quite frequently, so it would make sense to support it.

Related to:
https://github.com/CadQuery/CQ-editor/issues/31

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
This will resolve #118 

obraz

```python
import cadquery as cq 
b = cq.Workplane().box(10,10,1).faces('>Z').workplane()

e = b.polarArray(3.2,0,360,8,rotate=False).slot2D(3,1,90).cutThruAll()
show_object(e)

e = b.polarArray(3.2,0,360,8,rotate=True).slot2D(3,1,90).cutThruAll()
show_object(e.translate((12,0,0)))
```

### CODE REVIEW

File: cadquery/cq.py [hunk 1]
```diff
@@ -1376,12 +1385,19 @@ def polarArray(
 
         # Add additional elements
         for i in range(1, count):
-            phi = math.radians(startAngle + (angle * i))
-            x = radius * math.cos(phi)
-            y = radius * math.sin(phi)
-            points.append((x, y))
+            phi_deg = startAngle + (angle * i)
+            phi = math.radians(phi_deg)
+            x = radius * math.sin(phi)
```
Comment (jmwright) [inline cadquery/cq.py:1390]: Was the code wrong before when `x` was `cos` and `y` was `sin`?

File: tests/test_cadquery.py [hunk 2]
```diff
@@ -969,25 +969,42 @@ def testPolarArray(self):
...
+            lambda l: l.wrapped.Transformation().GetRotation().GetRotationAngle()
+            * 180.0
+            / math.pi
+        )
+
         # Test for proper placement when fill == True
         s = Workplane("XY").polarArray(radius, 0, 180, 3)
-        self.assertAlmostEqual(0, s.objects[1].x)
-        self.assertAlmostEqual(radius, s.objects[1].y)
+        self.assertAlmostEqual(0, to_y(s.objects[1]))
```
Comment (jmwright) [inline tests/test_cadquery.py:982]: `x` and `y` seem to be swapped from what they were in all these tests. Is that intended?

File: cadquery/cq.py (hunk 1 above)
Comment (adam-urbanczyk) [reply cadquery/cq.py:1390]: I changed the sign convention here: 0 angle maps to (0,r) and the points are distributed clockwise. I think that this is more intuitive but I'm open for discussion.

File: tests/test_cadquery.py (hunk 2 above)
Comment (adam-urbanczyk) [reply tests/test_cadquery.py:982]: Related to the comment above

File: cadquery/cq.py (hunk 1 above)
Comment (jmwright) [reply cadquery/cq.py:1390]: I'm fine with that, just wanted to be sure.

**Comment by jmwright** [comment]
@adam-urbanczyk Let me know whether you want to merge it, or whether I should.

**Comment by adam-urbanczyk** [comment]
Feel free to merge.

**Comment by jmwright** [comment]
Ok, merging.

**Pull request merged**